File: packages/domain/aquarium.ts
import { Entity } from "./entity";

export interface Aquarium extends Entity {
  id: string;
  name: string;
  liter: number;
  datum: Date;
}

export interface AquariumFormValues {
  name: string;
  liter: number;
  datum?: Date;
}

export default Aquarium;

File: Backend/Aqua.Domain/Aquarium.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace com.marcelbenders.Aqua.Domain;

public class Aquarium : IEntity
{
    public Guid Id { get; set; }

    public string UserId { get; set; }

    [ForeignKey("UserId")] 
    public virtual AppUser AppUser { get; set; }

    public string Name { get; set; }

    public int Liter { get; set; }

    public DateTimeOffset Datum { get; set; }
}